diff --git a/and_ivi/exynosautov920-evt0-idcevo-sp21-b1-android-vm.dts b/and_ivi/exynosautov920-evt0-idcevo-sp21-b1-android-vm.dts
index 870d142..dcb1847 100644
--- a/and_ivi/exynosautov920-evt0-idcevo-sp21-b1-android-vm.dts
+++ b/and_ivi/exynosautov920-evt0-idcevo-sp21-b1-android-vm.dts
@@ -114,17 +114,40 @@
 	status = "disabled";
 };
 
-&pcie_ep3 {
-	status = "disabled";
-	interrupts = <0 441 4>,
-			<0 418 4>,
-			<0 419 4>,
-			<0 420 4>,
-			<0 421 4>,
-			<0 414 4>,
-			<0 415 4>,
-			<0 416 4>,
-			<0 417 4>;
+#if 0
+&pinctrl_peric0 {
+	wifi_reg_ch2: wifi_reg_ch2 {
+		samsung,pins = "gpp1-5";
+		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+		samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+	};
+
+	wifi_reg_ch3: wifi_reg_ch3 {
+		samsung,pins = "gpp1-7";
+		samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+		samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+		samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+	};
+
+	wifibt_power_enable_pin_active: wifibt_power_enable_pin_active {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_UP>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+                //bias-pull-up;
+                //output-high;
+        };
+
+        wifibt_power_enable_pin_sleep: wifibt_power_enable_pin_sleep {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_DOWN>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_OUT0>;
+                //bias-disable;
+                //output-low;
+        };
 };
 
 &pcie_2 {
@@ -132,14 +155,76 @@
 	use-msi = "true";
 	use-bifurcation = "true";
 	num-lanes = <1>;
+	pinctrl-names = "default", "sleep";
+	//pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2>;
+	pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_active>;
+        pinctrl-1 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_sleep>;
+	gpio_wifi_reg_on = <&gpp1 5 GPIO_ACTIVE_LOW>;
 };
 
 &pcie_3 {
 	status = "okay";
-	interrupts = <0 441 4>, <0 418 4>;
 	use-msi = "true";
 	use-bifurcation = "true";
 	num-lanes = <1>;
+	pinctrl-names = "default", "sleep";
+	//pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3>;
+	pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3 &wifibt_power_enable_pin_active>;
+        pinctrl-1 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3 &wifibt_power_enable_pin_sleep>;
+	gpio_wifi_reg_on = <&gpp1 7 GPIO_ACTIVE_LOW>;
+};
+
+#endif
+
+&pinctrl_peric0 {
+        wifi_reg_ch2: wifi_reg_ch2 {
+                samsung,pins = "gpp1-5";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+        };
+
+        wifi_reg_ch3: wifi_reg_ch3 {
+                samsung,pins = "gpp1-7";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+        };
+
+        wifibt_power_enable_pin: wifibt_power_enable_pin {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+		samsung,pin-val = <1>;
+
+		samsung,pin-con-pdn = <EXYNOS_PIN_PDN_OUT0>;
+                samsung,pin-pud-pdn = <EXYNOS_PIN_PULL_DOWN>;		
+                //bias-pull-up;
+                //output-high;
+        };
+};
+
+&pcie_2 {
+        status = "okay";
+        use-msi = "true";
+        use-bifurcation = "true";
+        num-lanes = <1>;
+        pinctrl-names = "default";
+        //pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2>;
+        pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin>;
+        gpio_wifi_reg_on = <&gpp1 5 GPIO_ACTIVE_LOW>;
+};
+
+&pcie_3 {
+        status = "okay";
+        use-msi = "true";
+        use-bifurcation = "true";
+        num-lanes = <1>;
+        pinctrl-names = "default";
+        pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3>;
+        //pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3 &wifibt_power_enable_pin>;
+        gpio_wifi_reg_on = <&gpp1 7 GPIO_ACTIVE_LOW>;
 };
 
 /* Section to Enable Nodes */
diff --git a/and_ivi/exynosautov920-evt0-idcevo-sp25-b1-android-vm.dts b/and_ivi/exynosautov920-evt0-idcevo-sp25-b1-android-vm.dts
index 9f3d505..f074026 100644
--- a/and_ivi/exynosautov920-evt0-idcevo-sp25-b1-android-vm.dts
+++ b/and_ivi/exynosautov920-evt0-idcevo-sp25-b1-android-vm.dts
@@ -114,17 +114,39 @@
 	status = "disabled";
 };
 
-&pcie_ep3 {
-	status = "disabled";
-	interrupts = <0 441 4>,
-			<0 418 4>,
-			<0 419 4>,
-			<0 420 4>,
-			<0 421 4>,
-			<0 414 4>,
-			<0 415 4>,
-			<0 416 4>,
-			<0 417 4>;
+&pinctrl_peric0 {
+        wifi_reg_ch2: wifi_reg_ch2 {
+                samsung,pins = "gpp1-5";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+        };
+
+	wifi_reg_ch3: wifi_reg_ch3 {
+                samsung,pins = "gpp1-7";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+        };
+	
+	wifibt_power_enable_pin_active: wifibt_power_enable_pin_active {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+                bias-pull-up;
+                output-high;
+        };
+
+        wifibt_power_enable_pin_sleep: wifibt_power_enable_pin_sleep {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+                bias-disable;
+                output-low;
+        };
+
 };
 
 &pcie_2 {
@@ -132,14 +154,23 @@
         use-msi = "true";
         use-bifurcation = "true";
         num-lanes = <1>;
+	pinctrl-names = "default", "sleep";
+	//pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2>;
+	pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_active>;
+        pinctrl-1 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_sleep>;
+        gpio_wifi_reg_on = <&gpp1 5 GPIO_ACTIVE_HIGH>;
 };
 
 &pcie_3 {
 	status = "okay";
-	interrupts = <0 441 4>, <0 418 4>;
         use-msi = "true";
         use-bifurcation = "true";
         num-lanes = <1>;
+	pinctrl-names = "default", "sleep";
+	//pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3>;
+	pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3 &wifibt_power_enable_pin_active>;
+        pinctrl-1 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3 &wifibt_power_enable_pin_sleep>;
+        gpio_wifi_reg_on = <&gpp1 7 GPIO_ACTIVE_HIGH>;
 };
 
 /* Section to Enable Nodes */
diff --git a/and_ivi/exynosautov920-evt0-idcevo-sp25-phy-b1-android-vm.dts b/and_ivi/exynosautov920-evt0-idcevo-sp25-phy-b1-android-vm.dts
index eb07810..f5438dd 100644
--- a/and_ivi/exynosautov920-evt0-idcevo-sp25-phy-b1-android-vm.dts
+++ b/and_ivi/exynosautov920-evt0-idcevo-sp25-phy-b1-android-vm.dts
@@ -75,17 +75,38 @@
 	status = "disabled";
 };
 
-&pcie_ep3 {
-	status = "disabled";
-	interrupts = <0 441 4>,
-			<0 418 4>,
-			<0 419 4>,
-			<0 420 4>,
-			<0 421 4>,
-			<0 414 4>,
-			<0 415 4>,
-			<0 416 4>,
-			<0 417 4>;
+&pinctrl_peric0 {
+        wifi_reg_ch2: wifi_reg_ch2 {
+                samsung,pins = "gpp1-5";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+        };
+
+        wifi_reg_ch3: wifi_reg_ch3 {
+                samsung,pins = "gpp1-7";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+        };
+
+	wifibt_power_enable_pin_active: wifibt_power_enable_pin_active {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+                bias-pull-up;
+                output-high;
+        };
+
+        wifibt_power_enable_pin_sleep: wifibt_power_enable_pin_sleep {
+                samsung,pins = "gpp1-1";
+                samsung,pin-function = <EXYNOS_PIN_FUNC_OUTPUT>;
+                samsung,pin-pud = <EXYNOS_PIN_PULL_NONE>;
+                samsung,pin-drv = <EXYNOS5420_PIN_DRV_LV1>;
+                bias-disable;
+                output-low;
+        };
 };
 
 &pcie_2 {
@@ -93,14 +114,23 @@
         use-msi = "true";
         use-bifurcation = "true";
         num-lanes = <1>;
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_active>;
+        pinctrl-1 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_sleep>;
+	//pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2>;
+	gpio_wifi_reg_on = <&gpp1 5 GPIO_ACTIVE_HIGH>;
 };
 
 &pcie_3 {
 	status = "okay";
-	interrupts = <0 441 4>, <0 418 4>;
         use-msi = "true";
         use-bifurcation = "true";
         num-lanes = <1>;
+	pinctrl-names = "default", "sleep";
+	//pinctrl-0 = <&pcie_clkreq3 &pcie_perst3_out &wifi_reg_ch3>;
+	pinctrl-0 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_active>;
+        pinctrl-1 = <&pcie_clkreq2 &pcie_perst2_out &wifi_reg_ch2 &wifibt_power_enable_pin_sleep>;
+	gpio_wifi_reg_on = <&gpp1 7 GPIO_ACTIVE_HIGH>;
 };
 
 /* Section to Enable Nodes */
